packages/firebase_database/firebase_database/lib/ui/firebase_sorted_list.dart

```diff
@@ -89,9 +89,11 @@ class FirebaseSortedList extends ListBase<DataSnapshot>
   }
 
   void _onChildAdded(DatabaseEvent event) {
-    _snapshots.add(event.snapshot);
+    DataSnapshot dataSnapshot = event.snapshot;
+
+    _snapshots.add(dataSnapshot);
     _snapshots.sort(comparator);
-    onChildAdded!(_snapshots.indexOf(event.snapshot), event.snapshot);
+    onChildAdded!(_snapshots.indexOf(dataSnapshot), event.snapshot);
```

Comment:
this gives correct index

Reply:
I can't see how your changes make a difference 🤔.  But they can't hurt either, so I'll take your word for it.

Reply:
please check out the example given tickets.
both lines look similar but don't work the same when we have sorting.

Reply:
Isn't it hiding a race condition or something? Feels weird to me too

Reply:
A code like this is likely to get simplified by the next person going over this since there are no comments about why we create a local variable (apart from it works)

Also a test for this specific use case would probably need to be written to prevent regression in the future